assignments/lambda-classes.js

```diff
@@ -1 +1,133 @@
+        return `${student.name} receives a perfect score on ${subject}`;
+    }
+    randomGrade (student) {
+        student.grade = Math.round(Math.random()*100);
+        return `${student.name} your new score is %${student.grade}`
+    }
+}//Instructor class
+
+class Student extends Person {
+    constructor(StudentOptions) {
+    super(StudentOptions);
+    this.previousBackground = StudentOptions.previousBackground;
+    this.className = StudentOptions.className;
+    this.favSubjects = StudentOptions.favSubjects;
+    this.grade = StudentOptions.grade;
+    }
+    listsSubjects() {
+        for (let i = 0; i < this.favSubjects.length; i++) {
+            console.log(this.favSubjects[i]);
+        };
+        return `${this.favSubjects}`
+    }
+    PRAssignment(subject) {
+        return `${this.name} has submitted a PR for ${subject}`;
+    }
+    sprintChallenge(subject) {
+        return `${this.name} has begun sprint challenge on ${subject}`;
+    }
+    graduate() {
+        if (this.grade > 70) {
+            return `Congratulations you graduated, your grade is a ${this.grade}`;
+        } else {
+            return `Sorry, you failed. See if the Instructor or Project Manager will grade your assignments again.`
+        }
+    }
+}//Student class
+
+class ProjectManager extends Instructor {
+    constructor(PMOptions) {
+    super(PMOptions);
+    this.gradClassName = PMOptions.gradClassName;
+    this.favInstructor = PMOptions.favInstructor;
+    this.superPower = PMOptions.superPower;
+    }
+    standUP(channel) {
+        return `${this.name} announces to ${channel}, @channel standy time!`;
+    }
+    debugsCode(student, subject) {
+       return (`${this.name} debugs ${student.name}'s code on ${subject}`);
+    }
+}
+
+const josh = new Instructor ({
+    'name': 'Josh',
+    'location': 'Utah',
+    'age': '35',
+    'gender': 'M',
+    'specialty': 'UI',
+    'favLanguage': 'CSS',
+    'catchPhrase': 'hue hue'
+});
+
+const kaitlyn = new ProjectManager ({
+    'name': 'Kaitlyn',
+    'location': 'Texas',
+    'age': '29',
+    'gender': 'F',
+    'gradClassName': 'CS9',
+    'favInstructor': 'Beej',
+    'superPower': 'x-ray code vision',
+});
+
+let clint = new Student ({
+    'name': 'Clint',
+    'location': 'Idaho',
+    'age': '27',
+    'gender': 'M',
+    'previousBackground': 'Sourcing',
+    'className': 'CS 13',
+    'grade': '100',
+    'favSubjects': ['HTML', 'CSS', 'JavaScript'],
+});
+
+let newGuy = new Student ({
+    'name': 'Dude',
+    'location': 'Montana',
+    'age': '14',
+    'gender': 'M',
+    'grade': '100',
+    'favSubjects': ['Bears', 'Beets', 'Battle Star Gallactic', 'Michael']
+});
+
+console.log(josh.speak());
+console.log(kaitlyn.speak());
+console.log(clint.speak());
+console.log(josh.demo('the Banjo'));
+console.log(josh.grade(clint, "the Banjo"));
+console.log(clint.listsSubjects.call(clint, ...clint.favSubjects));
+console.log(clint.PRAssignment('the Banjo'));
```

Comment:
Now that your `listSubjects` method doesn't require any parameters, you can just do:
`console.log(clint.listsSubjects.call(clint);`